File: java/sourceafis/src/sourceafis/matching/minutia/PairInfo.java
package sourceafis.matching.minutia;

public class PairInfo implements Cloneable{
	public MinutiaPair pair;
    public MinutiaPair reference;
    public int supportingEdges;

    public Object clone()
    {
    	PairInfo pairInfo = new PairInfo();
    	if (this.pair != null)
    		pairInfo.pair = this.pair.clone();  
        pairInfo.supportingEdges = this.supportingEdges;
        if (this.reference != null)
        	pairInfo.reference = this.reference.clone();
        return pairInfo;
    }
}

File: SourceAFIS/Matching/Minutia/PairInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SourceAFIS.Dummy;

namespace SourceAFIS.Matching.Minutia
{
    public sealed class PairInfo : ICloneable
    {
        public MinutiaPair Pair;
        public MinutiaPair Reference;
        public int SupportingEdges;

        public object Clone()
        {
            return new PairInfo()
            {
                Pair = this.Pair,
                SupportingEdges = this.SupportingEdges,
                Reference = this.Reference
            };
        }
    }
}
